matrix_client/crypto/olm_device.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+import olm
+from canonicaljson import encode_canonical_json
+
+from matrix_client.checks import check_user_id
+
+logger = logging.getLogger(__name__)
+
+
+class OlmDevice(object):
```

Comment:
Is there a reason to make this class only handle crytpographic stuff? If not, it would be nice if this class was just called `Device`. I have a plan for at some point passing through all `MatrixHttpApi` methods that are related to specific class entity through to that class (when there isn't a higher-level function implementing the same behavior), so a Device class is something we'd need eventually for that.

Then again, there are benefits to all the crypto stuff living in its own module without it being polluted with other functionality. It probably is best to just plan on having that future `Device` class simply inherit from this one. Thoughts?

Reply:
Then again, there are benefits to all the crypto stuff living in its own module without it being polluted with other functionality. It probably is best to just plan on having that future `Device` class simply inherit from this one. Thoughts?

Reply:
This class ends up being 500 lines-ish of stuff that is a lot less trivial than what this PR lets show, so I'm pretty sure we don't want it to be something else than pure crypto. I even think that a completely separate Device class will be preferable when it gets to that.  
Also Device is an ambiguous name, because you'd expect it to be a general class for every devices (ours and all those we interact with), when in this case it is really only ours.

Reply:
Alright. You've convinced me.